monero/wallet_client_test.go

```diff
@@ -228,6 +228,44 @@ func TestCallGenerateFromKeys(t *testing.T) {
 	t.Logf("Address %s", addr.Address)
 }
 
+func TestCallGenerateFromKeys_UnusualAddress(t *testing.T) {
+	// this tests calling generateFromkeys passing an address derived in
+	// a non-standard manner (ie. the view keys aren't a hash of the spend
+	// key). the view key is actually a completely different key altogether.
```

Comment:
I'd move this comment above the function declaration, so it is a description of the test itself, and doesn't look like a description of the first part of the test. Also describe what the goal of the test is. Are we testing that `monero-wallet-rpc` generates the same address that we do?  That the wallet works (might need a transfer for that test to be complete).  Are we testing that `monero-wallet-rpc` accepts view keys that are not derived from the spend key when using [generate_from_keys](https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#generate_from_keys)?

Reply:
yeah, we're testing that `monero-wallet-rpc` accepts an address that doesn't match the private view key provided. will update!